punjab/session.py

```diff
@@ -83,8 +83,14 @@ def make_session(pint, attrs, session_type='BOSH'):
     if s.hostname in ['localhost', '127.0.0.1']:
         connect_srv = False
     if not connect_srv:
-        reactor.connectTCP(s.hostname, s.port,
-                           s, bindAddress=pint.bindAddress)
+        if s.port == 5223 and ssl:
```

Comment:
Checking this specific port seems a bit restrictive. What if you want to have ssl but use a different port?

Reply:
That's indeed more like a quick hack. I removed it from the PR for the time being until I might come up with a better implementation.